### PR TITLE
V0A-2306: Display 'Pending' status on upload page

### PR DESCRIPTION
**Pull request by valydia**

### CODE REVIEW

File: app/assets/javascripts/voabarxmlfrontend.js [hunk 1]
```diff
@@ -186,44 +186,86 @@ $(document).ready(function() {
...
+                    console.debug("scheduling ajax call, refreshUrl", refreshUrl)
+
+                    $.getJSON(refreshUrl, function (data, textStatus, jqXhr) {
+                        if (jqXhr.status === 200) {
+                            if ($("#reportStatus").val() !== data.status) {
+                                console.debug("status changed, updating page", data.status);
+
+                                if (data.status === "Failed" || data.status === "Submitted" || data.status === "Done") {
+                                    console.debug("Reached final status, removing refresh", status)
+                                    clearInterval(window.refreshIntervalId);
```
Comment (valydia) [inline app/assets/javascripts/voabarxmlfrontend.js:217]: Now we only redirect to the confirmation page, when reaching one of the final status

File: app/controllers/ConfirmationController.scala [hunk 2]
```diff
@@ -60,6 +60,19 @@ class ConfirmationController @Inject()(appConfig: FrontendAppConfig,
       }).valueOr(failPage => failPage)
   }
 
+  def onPageRefresh(reference: String) = getData.async {
```
Comment (valydia) [inline app/controllers/ConfirmationController.scala:63]: I ve just moved the method at the top of the file to comapre it with `onPageLoad` - they quite similar - we might be able to merge the 2 at some point

File: app/views/councilTaxUpload.scala.html [hunk 3]
```diff
@@ -29,6 +29,13 @@
 
 @(baCode: String, fileUploadForm: Form[_], initiateResponse: Option[InitiateResponse] = None)(implicit request: Request[_], messages: Messages)
 
+@label = {
+
+<h1 class="govuk-label-wrapper" id="heading">
```
Comment (valydia) [inline app/views/councilTaxUpload.scala.html:34]: Needed to add an id to the Label (to inject some html after it when needed), couldn't seem to do it with the api from `play-frontend-govuk`

File: app/views/councilTaxUpload.scala.html [hunk 4]
```diff
@@ -47,8 +54,9 @@
     @initiateResponse.map { ir =>
         <input type="hidden" id="councilTaxUploadEmptyFileError" value="@(routes.CouncilTaxUploadController.onPageLoad(true))" />
         <input type="hidden" id="councilTaxUploadReportError" value="@(routes.CouncilTaxUploadController.onError(ir.reference))" />
-        <input type="hidden" id="councilTaxUploadFormRedirect" value="@(routes.ConfirmationController.onPageLoad(ir.reference))" />
+        <input type="hidden" id="councilTaxUploadFormRedirect" value="@(routes.ConfirmationController.onPageRefresh(ir.reference))" />
```
Comment (valydia) [inline app/views/councilTaxUpload.scala.html:57]: When redirecting - using the `pageRefresh` which is similar to onPageLoad, except it return the status as well... - Might be able to delete onPageLoad, but I leave it for some tech debt

**Review by rhubner — APPROVED** [review]

**Pull request merged**